### PR TITLE
refactor: move func get_current_cluster_name out of duplication_common.h

### PR DESCRIPTION
**Pull request by levy5307**
`get_current_cluster_name` has nothing to do with duplication. So I move this function into common.h which is new created

### CODE REVIEW

File: include/dsn/dist/replication/common.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/*
```
Comment (hycdong) [inline include/dsn/dist/replication/common.h:1]: Why would you add a new file called `common.h`, not using `replication_common.h`, what's the difference between them?

Comment (Shuo-Jia) [reply include/dsn/dist/replication/common.h:1]: +1, add into https://github.com/XiaoMi/rdsn/blob/v2.3.0/src/common/replication_common.cpp is ok

Comment (levy5307) [reply include/dsn/dist/replication/common.h:1]: replication_common.h has a lot of classes and functions that used for replication. 
I don't want to put  funcget_current_cluster_name into it.

Comment (hycdong) [reply include/dsn/dist/replication/common.h:1]: `replication_common.h` is not only used for replica, but also used for meta, although its name is `replication_common`.
It has already been included in lots of meta header files such as `meta_service.h` and `server_state.h`.  I think `replication_common.h` and `common.h` are same.

Comment (levy5307) [reply include/dsn/dist/replication/common.h:1]: I said replication, not replica

Comment (levy5307) [reply include/dsn/dist/replication/common.h:1]: In my opinion, replication_common.h is used for replication, using by meta and replica.
common.h may be used in everywhere

Comment (hycdong) [reply include/dsn/dist/replication/common.h:1]: It's my fault that my pervious comment didn't show my opinion clearly, actually `replication_common.h` can also be used everywhere, such as `ddl_client`. And `replication_common.h` is actually the common, but its name is not suitable. I didn't say you shouldn't add a new file, I just don't figure out the difference.
If you insist on adding a new `common.h`, I think you should explain the difference between it and the original common file, or you should add your future refacion plan about how to solve the two files who have smiliar names.

Comment (levy5307) [reply include/dsn/dist/replication/common.h:1]: I think replication_common.h can't be used in everywhere. Just as it's name, it's the common file for replication. I think we used it in a wrong way.

Comment (hycdong) [reply include/dsn/dist/replication/common.h:1]: Okay, its name may lead to misunderstanding. How about adding some TODO comments in `replication_common.h` to explain that we plan to refactor it?

Comment (Shuo-Jia) [reply include/dsn/dist/replication/common.h:1]: > In my opinion, replication_common.h is used for replication, using by meta and replica.
> common.h may be used in everywhere

You has put the header into `dist/replication` that it do mean `replication_common`, so I think this file is same as `replication_common`.

if you want add `general file` to support `all common` function, that is `tools` , or you shouldn't put current module.

However, define `get_cluster_name` under `replication` module seem to be suitable.

Comment (GiantKing) [reply include/dsn/dist/replication/common.h:1]: If we add a new file, shell we refactor the code and correct the mistake we made bebore? Or we can change "replication_common.h" to "common.h" directly?

Comment (levy5307) [reply include/dsn/dist/replication/common.h:1]: I objected to your suggestion. You get misunderstanding about  `tools`. It's not a tool in rdsn.
And I'd rather put it out of directory `dist/replication`.

Comment (levy5307) [reply include/dsn/dist/replication/common.h:1]: > Okay, its name may lead to misunderstanding, and we should update its wrong usage. How about adding some TODO comments in `replication_common.h` to explain that we plan to refactor it?

I agree with you

Comment (levy5307) [reply include/dsn/dist/replication/common.h:1]: > If we add a new file, shell we refactor the code and correct the mistake we made bebore? Or we can change "replication_common.h" to "common.h" directly?

I have thought about your idea. But there are a lot of inner types in rdsn, which won't expose outside. If we do it like this, we should move `replication_common.h` under directory `include`, which would expose to other code.

**Review by hycdong — APPROVED** [review]

**Review by Shuo-Jia — APPROVED** [review]

**Pull request merged**